app/models/invoice.rb

```diff
@@ -9,4 +9,16 @@ class Invoice < ApplicationRecord
   def customer_name
     (customer.first_name) + " " + (customer.last_name)
   end
+
+  def merchant_items(merchant)
+    Item.joins(:invoice_items).where( items: {merchant_id: merchant.id})
```

Comment:
this doesn't seem to be referencing the current context of invoice. It seems to be calling all items from `Item`. Shouldn't it be `invoice.items.` to scope down to only this invoice's items?